test/server.cpp

```diff
@@ -197,6 +203,14 @@ TEST_F(ServerTest, 200)
     EXPECT_EQ(200, zfs1_->GET(res.url)->status) << "res.url: " << res.url;
 }
 
+// seperate test for 204 code
+
+TEST_F(ServerTest, 204)
+{
+  for ( const Resource& res : resources_204 )
+    EXPECT_EQ(204, zfs1_->GET(res.url)->status) << "res.url: " << res.url;
+}
+
```

Comment:
It doesn't make much sense to have multiple search URLs returning empty results (the reason for multiple URLs  in the other cases is to provide maximal coverage). This test can exercise a single URL similar to `ServerTest.SuccessfulSearchForAnArticleTitleRedirectsToTheArticle` (and you can rename it to `EmptySearchReturnsA204StatusCode`)